api/genomics/src/main/java/org/pmiops/workbench/genomics/RandomizeVcf.java

```diff
@@ -93,8 +92,8 @@ protected VariantContext randomizeVariant(VariantContext variant) {
     variantContextBuilder.alleles(variant.getAlleles());
 
     List<Genotype> randomizedGenotypes =
-        variant.getGenotypes().stream()
-            .map(genotype -> randomizeGenotype(variant, genotype))
+        IntStream.rangeClosed(1, this.samples)
+            .mapToObj(i -> randomizeGenotype(variant, variant.getGenotype(0), i))
```

Comment:
I don't understand a lot of the GATK models but I thought this part seemed a little suspicious since we're only ever grabbing the first genotype out of the variant. Is that ok?

Reply:
Yes. All we want to randomize is the alleles. We're starting from a single-sample vcf anyways so there will only ever be one genotype in the variant. I can specify that in the comments for the invocation and add a comment here, if you think that would help.